### ISSUE TITLE
allowJs option with namespaces

### ISSUE THREAD

**Issue by tsschaffert** [Microsoft/TypeScript#9474]
I am currently trying to put the compiler option "allowJs" to use. My goal is to convert an existing JavaScript application into TypeScript step by step. The application does not use (external) modules, so I'm converting the JavaScript namespaces into real TypeScript namespaces, but I'm running into "duplicate identifier" errors as outlined below. Shouldn't it be possible to combine TypeScript and JavaScript using the same namespace? Maybe I'm doing something wrong or this is not intended, but I expected this to work.

**TypeScript Version:**  1.8.10

**Code**
File 1:
```ts
namespace myapp {
    export var c1 = 5;
}
```
File 2:
```js
var myapp;
(function (myapp) {
    myapp.c2 = 4;
})(myapp || (myapp = {}));
```
tsconfig.json
```json
{
    "compilerOptions": {
        "allowJs": true,
        "outFile": "dist/appBundle.js"
    }
}
```

**Expected behavior:**
Bundling the compiled code of file 1 and the code of file 2 without modifications into a single JavaScript file.

**Actual behavior:** 
The compilation is done, but the compiler throws an error: "error TS2300: Duplicate identifier 'myapp'."

**Comment by basarat** [comment]
> Duplicate identifier 'myapp'

Personal opinino: Recommend using modules please (https://basarat.gitbooks.io/typescript/content/docs/tips/outFile.html) :rose:

**Comment by tsschaffert** [comment]
I agree that modules are a better solution in many cases, but as I am working on an existing application, I want to do the conversion one step at a time. 
Regardless, I still think that the option "allowJs" should at least somehow work when using namespaces.

**Comment by mhegazy** [comment]
@sandersn this is would be the next step after https://github.com/Microsoft/TypeScript/issues/16489. If we track the property assignment to a target that is a parameter of a an IIFE, we should redirect the declaration to be on the argument of the IIFE instead..
